server/src/main/java/org/apache/kafka/server/ClientMetricsManager.java

```diff
@@ -288,6 +307,9 @@ private ClientMetricsInstance createClientInstanceAndUpdateCache(Uuid clientInst
         ClientMetricsInstanceMetadata instanceMetadata) {
 
         ClientMetricsInstance clientInstance = createClientInstance(clientInstanceId, instanceMetadata);
+        // Maybe add client metrics, if metrics not already added. Metrics might be already added
+        // if the client instance was evicted from the cache because of size limit.
```

Comment:
Hmm, why would the metric be already added if the client was evicted? During eviction, we removed the metric.

Reply:
Hmm, if we evict a client instance from LRU cache, should we remove the corresponding metrics?

Reply:
The problem is that the eviction on size happens inside common LRUCache class where the code from this manager cannot be instrumented. I do not see any custom override that can be provided to LRUCache class. 

Having said that, there won't be any hanging additional instance metrics sensors when eviction happens on size as anyways time-based eviction will remove such instance metrics. Also, if the respective instance reports metrics prior time-based eviction, then the metrics numbers for that instance will be correctly reported as well. Hence, I do think we should not evict metrics sensors on size-based eviction.